README.md

```diff
@@ -7,38 +7,133 @@
+
+## Usage
+
+### `aesgcm`
+
+All [Web Push libraries](https://github.com/web-push-libs) support the "aesgcm" scheme, as well as Firefox 46+ and Chrome 50+. The app server includes its public key in the `Crypto-Key` HTTP header, the salt and record size in the `Encryption` header, and the encrypted payload in the body of the `POST` request.
+
+* The `Crypto-Key` header comprises one or more comma-delimited parameters. The first parameter must include a `dh` name-value pair, containing the sender's Base64url-encoded public key.
+* The `Encryption` header must include a `salt` name-value pair containing the sender's Base64url-encoded salt, and an optional `rs` pair specifying the record size.
+
+If the `Crypto-Key` header contains multiple keys, the sender must also include a `keyid` to match the encryption parameters to the key. The drafts have examples for [a single key without a `keyid`](https://tools.ietf.org/html/draft-ietf-webpush-encryption-04#section-5), and [multiple keys with `keyid`s](https://tools.ietf.org/html/draft-ietf-httpbis-encryption-encoding-02#section-5.6).
+
+**ecec** will extract the relevant parameters from the `Crypto-Key` and `Encryption` headers before decrypting the message. You don't need to parse the headers yourself.
+
+The other difference from `aes128gcm` is how `prkInfo`, `keyInfo`, and `nonceInfo` are determined:
+
+* `prkInfo` is the static string `"Content-Encoding: auth\0"`.
+* `keyInfo` is `"Content-Encoding: aesgcm\0P-256\0"`, followed by the length-prefixed (unsigned 16-bit integers) receiver and sender public keys in uncompressed form.
+* `nonceInfo` is `"Content-Encoding: nonce\0P-256\0"`, followed by the length-prefixed keys in the same form as `keyInfo`.
+
+```c
+int
+ece_aesgcm_decrypt(const ece_buf_t* rawRecvPrivKey, const ece_buf_t* authSecret,
+                   const char* cryptoKeyHeader, const char* encryptionHeader,
+                   const ece_buf_t* ciphertext, ece_buf_t* plaintext);
+```
+
+### `aes128gcm`
+
+This is the scheme from the latest version of the encrypted content-coding draft. It's not currently supported by any encryption library or browser, but will eventually replace `aesgcm`. This scheme removes the `Crypto-Key` and `Encryption` headers. Instead, the salt, record size, and sender public key are included in the payload as a binary header block.
+
+Additionally:
```

Comment:
Same as above. Not really sure that folks care about data that they can't change.